lib/sinon/extend.js

```diff
@@ -54,13 +52,12 @@ var hasDontEnumBug = (function () {
  *
  * Returns the extended target
  */
-function extend(target /*, sources */) {
+module.exports = function extend(target /*, sources */) {
     var sources = Array.prototype.slice.call(arguments, 1);
-    var source, i, prop;
-
-    for (i = 0; i < sources.length; i++) {
-        source = sources[i];
+    var i = 0;
+    var source, prop;
 
+    while ((source = sources[i++])) {
```

Comment:
Agreed, I find the change to be for the worse regarding readability. If it were to be changed for fewer variables I'd much more prefer to simply use a `sources.forEach()` construct, but that would leave ES3 browsers out in the cold. So just keep the existing variant.